internal/host/transition.go

```diff
@@ -112,38 +112,35 @@ func (th *transitionHandler) PostRegisterDuringReboot(sw stateswitch.StateSwitch
 		return errors.New("PostRegisterDuringReboot invalid argument")
 	}
 
-	if sHost.host.InstallationDiskPath == "" || sHost.host.Inventory == "" {
+	host := sHost.host
+	if host.InstallationDiskPath == "" || host.Inventory == "" {
 		return errors.New(fmt.Sprintf("PostRegisterDuringReboot host %s doesn't have installation_disk_path or inventory", *sHost.host.ID))
 	}
-
-	var installationDisk *models.Disk = nil
-
-	var inventory models.Inventory
-	err := json.Unmarshal([]byte(sHost.host.Inventory), &inventory)
+	installationDisk, err := hostutil.GetInventoryDisk(host, host.InstallationDiskPath)
 	if err != nil {
 		return errors.New(fmt.Sprintf("PostRegisterDuringReboot Could not parse inventory of host %s", *sHost.host.ID))
 	}
-
-	for _, disk := range inventory.Disks {
-		if hostutil.GetDeviceFullName(disk.Name) == sHost.host.InstallationDiskPath {
-			installationDisk = disk
-			break
-		}
-	}
-
 	if installationDisk == nil {
 		return errors.New(fmt.Sprintf("PostRegisterDuringReboot Could not find installation disk %s for host %s",
 			sHost.host.InstallationDiskPath, *sHost.host.ID))
 	}
 
-	statusInfo := fmt.Sprintf("Expected the host to boot from disk, but it booted the installation image - please reboot and fix boot order to boot from disk %s",
-		sHost.host.InstallationDiskPath)
-
-	if installationDisk != nil && installationDisk.Serial != "" {
-		statusInfo += fmt.Sprintf(" (%s)", installationDisk.Serial)
+	var statusInfo strings.Builder;
+	statusInfo.WriteString("Expected the host to boot from disk, but it booted the installation image - please reboot and fix boot order to boot from disk")
+	if installationDisk.Model != "" {
+		statusInfo.WriteString(installationDisk.Model)
+	}
+	if installationDisk.Serial != "" {
+		statusInfo.WriteString(" ")
```

Comment:
This space might be unnecessary?

If you're looking for multiple strings, space separated, with some of them potentially missing, I'd suggest just initializing an empty string array, appending to it, then using `strings.Join(' ', myStrings)`

Reply:
done